homeassistant/components/utility_meter/sensor.py

```diff
@@ -491,6 +491,7 @@ def async_reading(self, event: EventType[EventStateChangedData]) -> None:
             # If net_consumption is off, the adjustment must be non-negative
             self._state += adjustment  # type: ignore[operator] # self._state will be set to by the start function if it is None, therefore it always has a valid Decimal value at this line
 
+        self._unit_of_measurement = new_state.attributes.get(ATTR_UNIT_OF_MEASUREMENT)
```

Comment:
Should we warn when it was previously `None`. That seems like it would be a bug in the source integration

Reply:
Maybe something like https://github.com/home-assistant/core/blob/11cecc3f0a11e019315b8c2e35260065dd0c8556/homeassistant/components/sensor/__init__.py#L497

Reply:
I could not reproduce, but from the linked issue, the same source would result in different results... I'm guessing it's an issue while the system is starting and unit is not immediately available.